packages/storage-plus/README.md

```diff
@@ -384,7 +384,172 @@ fn demo() -> StdResult<()> {
+    Box::new(v.into_iter())
+  }
+}
+
+pub fn tokens<'a>() -> IndexedMap<'a, &'a str, TokenInfo, TokenIndexes<'a>> {
+  let indexes = TokenIndexes {
+    owner: MultiIndex::new(
+      |d, k| (Vec::from(d.owner.as_ref()), k),
+      "tokens",
+      "tokens__owner",
+    ),
+  };
+  IndexedMap::new("tokens", indexes)
+}
+```
+
+Let's discuss this piece by piece:
+```rust
+pub struct TokenIndexes<'a> {
+  // pk goes to second tuple element
+  pub owner: MultiIndex<'a, (Vec<u8>, Vec<u8>), TokenInfo>,
+}
+```
+
+These are the index definitions. Here there's only one index, called `owner`. There could be more, as public
+members of the `TokenIndexes` struct.
+
+We see that the `owner` index is a `MultiIndex`. A multi-index can have repeated values as keys. That's why
+the primary key is being added as the last element of the multi-index key.
+Like the name implies, this is an index over tokens, by owner. Given that an owner can have multiple tokens,
+we need a `MultiIndex` to be able to list / iterate over all the tokens a given owner has.
+
+So, to recap, the `TokenInfo`  data will originally be stored by `token_id` (which is a string value).
+You can see this in the token creation code:
+```rust
+    tokens().update(deps.storage, &msg.token_id, |old| match old {
+        Some(_) => Err(ContractError::Claimed {}),
+        None => Ok(token),
+    })?;
+```
+(Incidentally, this is using `update` instead of `save`, to avoid overwriting an already existing token).
+
+Then, it will be indexed by token `owner` (which is an `Addr`), so that we can list all the tokens an owner has.
+That's why the `owner` index key is `(Vec<u8>, Vec<u8>)`. The first owned element is the `owner` data
+(converted to `Vec<u8>`), whereas the second one is the `token_id` (also converted to `Vec<u8>`).
+
+The important thing here is that the key (and its components, in the case of a combined key) must implement
+the `PrimaryKey` trait. You can see that both the 2-tuple `(_, _)` and `Vec<u8>` do implement `PrimaryKey`:
+
+```rust
+impl<'a, T: PrimaryKey<'a> + Prefixer<'a>, U: PrimaryKey<'a>> PrimaryKey<'a> for (T, U) {
+    type Prefix = T;
+    type SubPrefix = ();
+
+    fn key(&self) -> Vec<&[u8]> {
+        let mut keys = self.0.key();
+        keys.extend(&self.1.key());
+        keys
+    }
+}
+```
+
+```rust
+impl<'a> PrimaryKey<'a> for Vec<u8> {
+    type Prefix = ();
+    type SubPrefix = ();
+
+    fn key(&self) -> Vec<&[u8]> {
+        vec![&self]
+    }
+}
+```
+
+---
+
+We can now see how it all works, taking a look at the remaining code:
+
+```rust
+impl<'a> IndexList<TokenInfo> for TokenIndexes<'a> {
+    fn get_indexes(&'_ self) -> Box<dyn Iterator<Item = &'_ dyn Index<TokenInfo>> + '_> {
+        let v: Vec<&dyn Index<TokenInfo>> = vec![&self.owner];
+        Box::new(v.into_iter())
+    }
+}
+```
+
+This implements the `IndexList` trait for `TokenIndexes`. This kind of serves two purposes (which are really
```

Comment:
You can mention this is more or less boiler-plate and needed for the internals. Do not try to customize this, just return a list of all indexes